bot/telegram_bot.py

```diff
@@ -83,6 +84,27 @@ async def stats(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
                      f"💰 For a total amount of ${cost_month:.2f}"
         await update.message.reply_text(usage_text)
 
+    async def resend(self, update: Update, context: ContextTypes.DEFAULT_TYPE):
+        """
+        Resend the last request
+        """
+        if not await self.is_allowed(update):
+            logging.warning(f'User {update.message.from_user.name} is not allowed to resend the message')
+            await self.send_disallowed_message(update, context)
+            return
+
+        chat_id = update.effective_chat.id
+        if chat_id not in self.last_message:
+            logging.warning(f'User {update.message.from_user.name} does not anything to resend')
+            await context.bot.send_message(chat_id=chat_id, text="You have nothing to resend")
+            return
+
+        # Update message text and send the req to promt
+        logging.info(f'Resending the last message to the user: {update.message.from_user.name}')
+        with update.message._unfrozen() as message:
+            message.text = self.last_message[chat_id]
```

Comment:
I'm not super stoked about using a private api here, but it does seem to be the easiest way to mutate the update. A cleaner approach would be to split the prompt method up such that we don't have to pass a full Update object here. @n3d1117 what do you think?